### PR TITLE
Add macro RelationIsNonblockRelation to expand code path like AO/CO

### PR DESCRIPTION
**Pull request by gfphoenix78**
The builtin types of tables, in kernel, are standard heap table and AO/CO. The custom table AM will fall one of the two code paths. RelationIsAppendOptimized will only choose the builtin AO/CO relations. Some custom table AM, like PAX, will run the same code path as AO/CO.

RelationIsNonblockRelation looks replacable by `!RelationIsHeap`, but they have different meanings. RelationIsNonblockRelation expand the relation type to run the code path like AO/CO. `!RelationIsHeap` emphasizes NOT heap relation.

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

File: src/include/utils/rel.h [hunk 1]
```diff
@@ -516,6 +517,23 @@ typedef struct ViewOptions
...
+ * they have different meanings. RelationIsNonblockRelation expand the
+ * relation type to run the code path like AO/CO. `!RelationIsHeap`
+ * emphasizes NOT heap relation.
+ *
+ * RelationIsNonblockRelation
+ *      True iff relation(table) should run the code path as AO/CO
+ */
+#define RelationIsNonblockRelation(relation) \
+	((relation)->rd_tableam && (relation)->rd_rel->relam != HEAP_TABLE_AM_OID)
+
```
Comment (avamingli) [inline src/include/utils/rel.h:536]: Hi, understand the change. 
 I have a idea that: do we have a certain PAX_AM_OID now? If true, it seems more reasonable to define RelationIsNonblockRelation to is PAX_AM_OID || RelationIsAppendOptimized.
Others LGTM

**Review by avamingli — APPROVED** [review]

**Review by avamingli — APPROVED** [review]

File: src/include/utils/rel.h (hunk 1 above)
Comment (gfphoenix78) [reply src/include/utils/rel.h:536]: PAX_AM_OID is defined in the pax code, the kernel should not depend on the oid of PAX.
We should enchance the macro if the context is changed.

File: src/include/utils/rel.h [hunk 2]
```diff
@@ -516,6 +517,29 @@ typedef struct ViewOptions
...
+ * they have different meanings. RelationIsNonblockRelation expand the
+ * relation type to run the code path like AO/CO. `!RelationIsHeap`
+ * emphasizes NOT heap relation.
+ *
+ * RelationIsNonblockRelation
+ *      True iff relation(table) should run the code path as AO/CO
+ */
+#define RelationIsNonblockRelation(relation) \
+	(RelationIsAppendOptimized(relation) || \
+	 (relation)->rd_rel->relam == PAX_AM_OID)
```
Comment (avamingli) [inline src/include/utils/rel.h:541]: add an extra brackets outside()?
	((RelationIsAppendOptimized(relation) || \
	 (relation)->rd_rel->relam == PAX_AM_OID))

**Review by avamingli — APPROVED** [review]
LGTM

File: src/include/utils/rel.h (hunk 2 above)
Comment (gfphoenix78) [reply src/include/utils/rel.h:541]: no, it's needless.

**Pull request merged**